### PR TITLE
WT-5722 Add standard name checks to incremental backup identifiers.

### PR DESCRIPTION
**Pull request by sueloverso**
@ddanderson and @keitharnoldsmith given yesterday's discussion about namespaces I have implemented changes to apply the same naming restrictions and checks to backup identifiers as we do to other names.

### CODE REVIEW

**Comment by ddanderson** [comment]
I think I may be missing something here.  If we're quoting the backup ids in the metadata file, why do we care if the caller has used a special character in it?  And why is "WiredTiger...." an illegal name for a backup id?  Will we want to use that portion of the backup id name space in the future?

**Comment by sueloverso** [comment]
@keithbostic felt it was cleaner and less likely to lead to bugs in any future to have identifiers stored in the metadata follw the same name-checked rules. I agree and I don't think that MongoDB is worried about a slightly restriced namespace since it is consistent with other naming.

File: src/schema/schema_util.c [hunk 1]
```diff
@@ -110,6 +110,33 @@ __wt_schema_session_release(WT_SESSION_IMPL *session, WT_SESSION_IMPL *int_sessi
...
+    /*
+     * Disallow JSON quoting characters -- the config string parsing code supports quoted strings,
+     * but there's no good reason to use them in names and we're not going to do the testing.
+     */
+    if (strpbrk(name, "{},:[]\\\"'") != NULL)
+        WT_RET_MSG(session, EINVAL,
+          "%s: WiredTiger objects should not include grouping "
+          "characters in their names",
+          name);
+    return (0);
```
Comment (keithbostic) [inline src/schema/schema_util.c]: Might as well join the `WT_RET_MSG` lines?

Comment (sueloverso) [reply src/schema/schema_util.c]: clang-format did that.

**Pull request merged**